src/ErrorFormatter/AbstractFormatter.php

```diff
@@ -0,0 +1,60 @@
+<?php
+declare(strict_types = 1);
+
+namespace Middlewares\ErrorFormatter;
+
+use Middlewares\Utils\Factory;
+use Middlewares\Utils\HttpErrorException;
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Throwable;
+
+abstract class AbstractFormatter implements FormatterInterface
+{
+    protected $responseFactory;
+    protected $contentTypes = [];
+
+    public function __construct(
+        ResponseFactoryInterface $responseFactory = null
+    ) {
+        $this->responseFactory = $responseFactory ?? Factory::getResponseFactory();
+    }
+
+    public function isValid(Throwable $error, ServerRequestInterface $request): bool
+    {
+        $accept = $request->getHeaderLine('Accept');
+
+        foreach ($this->contentTypes as $type) {
+            if (stripos($accept, $type) !== false) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    abstract protected function format(Throwable $error): string;
+
+    public function handle(Throwable $error, ServerRequestInterface $request): ResponseInterface
+    {
+        $response = $this->responseFactory->createResponse($this->errorStatus($error));
+        $response = $response->withHeader('Content-Type', $this->contentTypes[0]);
```

Comment:
There is no way to ensure this is correct. If a formatter has types `application/json` and `application/vnd.api+json` and the request has `Accept: application/vnd.api+json` then the wrong content type will be served.